### PR TITLE
base: lmp: provision TUF root metadata

### PR DESCRIPTION
**Pull request by mike-sul**
Fetch and store root metadata on an ostree-based rootfs during bitbaking.

### CODE REVIEW

**Comment by mike-sul** [comment]
This PR goes together with https://github.com/foundriesio/ci-scripts/pull/202

**Review by doanac — COMMENTED** [review]
lgtm

**Review by doanac — COMMENTED** [review]
looking good

File: meta-lmp-base/classes/lmp.bbclass [hunk 1]
```diff
@@ -11,6 +11,19 @@ include conf/machine/include/lmp-partner-custom.inc
 # Allow customizations per factory level
 include conf/machine/include/lmp-factory-custom.inc
 
+provision_root_meta () {
+       if [ -f "${SOTA_TUF_ROOT_FETCHER}" ] && [ -x "${SOTA_TUF_ROOT_FETCHER}" ]; then
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:15]: These variables are not defined with a standard value in normal lmp environment, so I would add a check if the variable is empty or valid as well.

**Comment by ricardosalveti** [comment]
Please squash the ak-lite update commits and remove draft status, as it seems we're close to get this merged.

**Comment by ricardosalveti** [comment]
Just noticed this is on top of https://github.com/foundriesio/meta-lmp/pull/731, so forget about my last comment.

**Comment by ricardosalveti** [comment]
Please rebase :-)

**Review by quaresmajose — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by Tim-Anderson — APPROVED** [review]
LGTM

File: meta-lmp-base/classes/lmp.bbclass [hunk 2]
```diff
@@ -11,6 +12,23 @@ include conf/machine/include/lmp-partner-custom.inc
...
+	if [ -n "${SOTA_TUF_ROOT_FETCHER}" ]; then
+		if [ -f "${SOTA_TUF_ROOT_FETCHER}" ] && [ -x "${SOTA_TUF_ROOT_FETCHER}" ]; then
+			bbplain "Fetching TUF root role metadata to the ostree-based rootfs..."
+
+			FACTORY="${LMP_DEVICE_FACTORY}" \
+			SOTA_TUF_ROOT_DIR="${SOTA_TUF_ROOT_DIR}" \
+			LOG_FILE="${SOTA_TUF_ROOT_LOG_FILE}" \
+				${SOTA_TUF_ROOT_FETCHER}
+		else
+			bbfatal "Provisioning of TUF root role metadata is turned ON, but the specified metadata fetcher is either doesn't exist or is not executable: ${TUF_ROOT_FETCHER}"
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:25]: Should be SOTA_TUF_ROOT_FETCHER instead of TUF_ROOT_FETCHER, right?

Comment (mike-sul) [reply meta-lmp-base/classes/lmp.bbclass]: Indeed, updated.

**Pull request merged**